### ISSUE TITLE
Bug: false negative in eslint/no-loss-of-precision

### ISSUE THREAD

**Issue by camc314** [eslint/eslint#19810]
### Environment

N/A as I reproduced this in the playground

### What parser are you using?

Default (Espree)

### What did you do?

```
/*eslint no-loss-of-precision: "error"*/

const one_before = 1000000000000000127
const this_shuold_error = 1000000000000000128
const one_after = 1000000000000000129
```
In the above code, `1000000000000000128` is not reporting an error despite it being truncated to `1000000000000000100` at runtime

### What did you expect to happen?

I expected an error to be reported on the `1000000000000000128` numeric literal

### What actually happened?

No error was reported

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tbG9zcy1vZi1wcmVjaXNpb246IFwiZXJyb3JcIiovXG5cbmNvbnN0IG9uZV9iZWZvcmUgPSAxMDAwMDAwMDAwMDAwMDAwMTI3XG5jb25zdCB0aGlzX3NodW9sZF9lcnJvciA9IDEwMDAwMDAwMDAwMDAwMDAxMjhcbmNvbnN0IG9uZV9hZnRlciA9IDEwMDAwMDAwMDAwMDAwMDAxMjkiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Originally flagged as a false positive in oxc, but it appears to be a bug in eslint instead > https://github.com/oxc-project/oxc/issues/11436
